src/js/system.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2022, Tony Atkins
+ * Copyright 2011-2020, Colin Clark
+ *
+ *  Licensed under the MIT license, see LICENSE for details.
+ */
+(function (fluid) {
+    "use strict";
+    var youme = fluid.registerNamespace("youme");
+
+
+    /**
+     * TODO: Make this grade something that is instantiated once and which can be referred to by any components that
+     * need it, such as by using a `fluid.resolveRootSingle` grade:
+     *
+     * https://docs.fluidproject.org/infusion/development/contexts#global-components-fluidresolveroot-and-fluidresolverootsingle
+     *
+     * Unfortunately this doesn't work as a named sub-component. We could instantiate an instance globally like the old
+     * `flock.enviro`.
+     *
+     */
+
+    /**
+     *
+     * Represents the overall Web MIDI system, including references to all the available MIDI ports and the MIDIAccess
+     * object.
+     *
+     */
+    fluid.defaults("youme.system", {
+        gradeNames: ["fluid.modelComponent", "fluid.resourceLoader"],
+
+        // Since Chrome always prompts regardless, we ask for permission to send sysex by default.
+        // https://chromestatus.com/feature/5138066234671104
+        sysex: true,
+        software: true,
+
+        resources: {
+            access: {
+                promiseFunc: "{that}.requestAccess"
+            }
+        },
+
+        model: {
+            access: "{that}.resources.access.parsed",
```

Comment:
Should explain what the resulting type of this is, is it the native MIDIAccess object?

Reply:
I added a comment.